sigmastate/src/main/scala/sigmastate/Values.scala

```diff
@@ -1522,4 +1523,121 @@ object Values {
       withSegregation(DefaultHeader, prop)
   }
 
+  case class Parameter private[sigmastate](
+    name: String,
+    description: String,
+    placeholder: Int
+  ) {
+    def serializeInto(w: SigmaByteWriter): Unit = {
+      ParameterSerializer.serialize(this, w)
+    }
+  }
+
+  object Parameter {
+    /** Immutable empty IndexedSeq, can be used to save allocations in many places. */
+    val EmptySeq: IndexedSeq[Parameter] = Array.empty[Parameter]
+  }
+
+
+  /** This is alternative representation of ContractTemplate when it cannot be parsed
+    * due to `error`. */
+  case class UnparsedContractTemplate(bytes: mutable.WrappedArray[Byte], error: ValidationException)
```

Comment:
I don't think we need to represent unparsable templates. This feature in ErgoTree was added because deserialization there should always succeed to implement soft-fork logic. Here it is not necessary.

Reply:
Sure, removed that part from the code.